soh/soh/OTRGlobals.cpp

```diff
@@ -1789,7 +1789,22 @@ extern "C" int CustomMessage_RetrieveIfExists(GlobalContext* globalCtx) {
             messageEntry = Randomizer_GetNaviMessage();
         } else if (Randomizer_GetSettingValue(RSK_SHUFFLE_MAGIC_BEANS) && textId == TEXT_BEAN_SALESMAN) {
             messageEntry = CustomMessageManager::Instance->RetrieveMessage(Randomizer::merchantMessageTableID, TEXT_BEAN_SALESMAN);
-        } else if (Randomizer_GetSettingValue(RSK_BOMBCHUS_IN_LOGIC) &&
+        } else if (Randomizer_GetSettingValue(RSK_SHUFFLE_MERCHANTS)) {
```

Comment:
This might cause issues for all the branches below this while this setting is on. The if statement will always enter this branch while the Shuffle Merchant setting is on, and then do nothing unless it happens to be one of those three, and then skip right to the end. I think you'll need to add a check for one of these 3 specific text IDs in the conditional for this else if branch.

Reply:
Pretty sure this also replaces nearly every single textbox in the game when this setting is on. The text IDs definitely need to be included in this if statement.